tools/crmadmin.c

```diff
@@ -215,19 +223,26 @@ do_work(void)
     xmlNode *msg_data = NULL;
     gboolean all_is_good = TRUE;
 
-    if (DO_HEALTH == TRUE) {
+    if ((DO_HEALTH == TRUE) || (DO_PACEMAKERD_HEALTH == TRUE)) {
         crm_trace("Querying the system");
 
         sys_to = CRM_SYSTEM_DC;
 
-        if (dest_node != NULL) {
+        if ((DO_HEALTH == TRUE) && (dest_node != NULL)) {
             sys_to = CRM_SYSTEM_CRMD;
             crmd_operation = CRM_OP_PING;
 
             if (BE_VERBOSE) {
                 expected_responses = 1;
             }
 
+        } else if (DO_PACEMAKERD_HEALTH == TRUE) {
+            sys_to = CRM_SYSTEM_MCP;
+            crmd_operation = CRM_OP_PING;
```

Comment:
Agreed that we need public API functions for interacting with pacemakerd. They can go in libcrmcommon. We always want an abstraction layer between external callers and IPC XML so the latter is internal implementation.

BTW @clumens has a project to create an API for the controller which would simplify crmadmin (and crm_node) further. There's quite a bit in the queue ahead of that though.